src/Pages/Repositories/ContentListTable/components/SnapshotListModal/DeleteSnapshotsModal/DeleteSnapshotsModal.tsx

```diff
@@ -0,0 +1,356 @@
+import { useQueryClient } from 'react-query';
+import { useHref, useLocation, useNavigate, useParams } from 'react-router-dom';
+import { useSnapshotListOutletContext } from '../SnapshotListModal';
+import useRootPath from 'Hooks/useRootPath';
+import { REPOSITORIES_ROUTE, TEMPLATES_ROUTE } from 'Routes/constants';
+import { SnapshotItem } from 'services/Content/ContentApi';
+import {
+  CONTENT_LIST_KEY,
+  LIST_SNAPSHOTS_KEY,
+  useBulkDeleteSnapshotsMutate,
+  useGetSnapshotList,
+} from 'services/Content/ContentQueries';
+import { TemplateItem } from 'services/Templates/TemplateApi';
+import {
+  GET_TEMPLATES_KEY,
+  useFetchTemplatesForSnapshots,
+} from 'services/Templates/TemplateQueries';
+import { formatDateDDMMMYYYY } from 'helpers';
+import ChangedArrows from '../components/ChangedArrows';
+import { SnapshotDetailTab } from '../../SnapshotDetailsModal/SnapshotDetailsModal';
+
+const useStyles = createUseStyles({
+  description: {
+    paddingTop: '12px', // 4px on the title bottom padding makes this the "standard" 16 total padding
+    color: global_Color_100.value,
+  },
+  removeButton: {
+    marginRight: '36px',
+    transition: 'unset!important',
+  },
+  link: {
+    padding: 0,
+  },
+  templateColumnMinWidth: {
+    minWidth: '200px!important',
+  },
+  expandableSectionMargin: {
+    marginTop: '8px',
+  },
+});
+
+export default function DeleteSnapshotsModal() {
+  const classes = useStyles();
+  const navigate = useNavigate();
+  const queryClient = useQueryClient();
+  const rootPath = useRootPath();
+  const path = useHref('content');
+  const pathname = path.split('content')[0] + 'content';
+  const { repoUUID: uuid = '' } = useParams();
+  const { search } = useLocation();
+  const [isLoading, setIsLoading] = useState(true);
+  const [affectsTemplates, setAffectsTemplates] = useState(false);
+  const [expandState, setExpandState] = useState({});
+  const maxTemplatesToShow = 3;
+
+  const {
+    clearCheckedSnapshots,
+    deletionContext: { checkedSnapshots },
+  } = useSnapshotListOutletContext();
+
+  const snapshotUUIDFromPath = new URLSearchParams(search).get('snapshotUUID')?.split(',');
+  const checkedSnapshotsArray = checkedSnapshots ? Array.from(checkedSnapshots) : [];
+  const uuids = snapshotUUIDFromPath?.length
+    ? snapshotUUIDFromPath
+    : checkedSnapshotsArray.length
+      ? checkedSnapshotsArray
+      : [];
+  const snapshotsToDelete = new Set(uuids);
+
+  const { mutateAsync: deleteSnapshots, isLoading: isDeletingSnapshots } =
+    useBulkDeleteSnapshotsMutate(queryClient, uuid, snapshotsToDelete);
+
+  const onClose = () => navigate(`${rootPath}/${REPOSITORIES_ROUTE}/${uuid}/snapshots`);
+
+  const onSave = async () => {
+    deleteSnapshots(snapshotsToDelete).then(() => {
+      onClose();
+      clearCheckedSnapshots();
+      queryClient.invalidateQueries(CONTENT_LIST_KEY);
+      queryClient.invalidateQueries(GET_TEMPLATES_KEY);
+      queryClient.invalidateQueries(LIST_SNAPSHOTS_KEY);
```

Comment:
Invalidations should be done in the useBulkDeleteSnapshotsMutate hook.
Is there a reason to add these here instead of there?

Reply:
right, don't know why I have split those, possibly for testing, moved to the delete hook 👍🏼